File: 007/main.c
#include <stdio.h>

int reverse(int x) {
    long long INT_MAX = 2147483647;
    long long INT_MIN = -2147483648;
    int positive = x >= 0 ? 1 : -1;
    long long ori = x >= 0 ? x : -1 * (long long)x;
    long long ans = 0;
    while (ori) {
        ans = ans * 10 + ori % 10;
        ori /= 10;
    }
    ans = positive * ans;
    if (positive > 0 && ans >= INT_MAX) {
        return 0;
    } else if (positive < 0 && ans <= INT_MIN) {
        return 0;
    }
    return ans;
}

int main(int argc, char const *argv[]) {
    printf("%d\n", reverse(1230));
    return 0;
}

File: 007/main.js
const INT_MAX = Math.pow(2, 31) - 1
const INT_MIN = -Math.pow(2, 31)

/**
 * @param {number} x
 * @return {number}
 */
var reverse = function (x) {
  let positive = x >= 0 ? 1 : -1
  let ori = x >= 0 ? x : -x
  ori = ('' + ori).split('')
  ori = ori.reverse().join('')
  ori = positive * parseInt(ori)
  if (ori >= INT_MAX || ori <= INT_MIN) {
    return 0
  }
  return ori
}

if (process.env.LZS) { // local test
  const assert = require('chai').assert
  assert.strictEqual(reverse(123), 321)
  assert.strictEqual(reverse(-123), -321)
  assert.strictEqual(reverse(0), 0)
  assert.strictEqual(reverse(10), 1)
  assert.strictEqual(reverse(-10), -1)
  assert.strictEqual(reverse(214748364), 463847412)
  assert.strictEqual(reverse(1111111122), 0)
  assert.strictEqual(reverse(-1111111122), 0)
}
